File: app/lib/domain/entities/episode.dart
class Episode {
  const Episode({
    required this.id,
    required this.name,
    required this.airDate,
    required this.episodeCode,
  });

  final int id;
  final String name;
  final String airDate;
  final String episodeCode;

  factory Episode.fromJson(Map<String, dynamic> json) {
    return Episode(
      id: json['id'] as int,
      name: json['name'] as String,
      airDate: json['airDate'] as String,
      episodeCode: json['episodeCode'] as String,
    );
  }
}

File: web/src/lib/types/episode.ts
export interface Episode {
  id: number;
  name: string;
  airDate: string;
  episodeCode: string;
}

export interface EpisodesPage {
  episodes: Episode[];
  page: number;
  totalPages: number;
  totalCount: number;
  hasNext: boolean;
  hasPrevious: boolean;
}

export interface Character {
  id: number;
  name: string;
  image: string;
}

export interface EpisodeDetail extends Episode {
  characters: Character[];
}
